src/Veldrid/Vk/VulkanUtil.cs

```diff
@@ -19,7 +19,7 @@ public static void CheckResult(VkResult result)
             }
         }
 
-        public static uint FindMemoryType(VkPhysicalDeviceMemoryProperties memProperties, uint typeFilter, VkMemoryPropertyFlags properties)
+        public static uint? TryFindMemoryType(VkPhysicalDeviceMemoryProperties memProperties, uint typeFilter, VkMemoryPropertyFlags properties)
```

Comment:
Should probably return a bool if you want to name the function tryXXX. (Similar to dictionary TryXXX functions). Also, use *out* parameter to return the actual result.

Reply:
I see your point, make sense.

Yeah, FindMemoryType sounds better than TryFindMemoryType but that’s personal preference I think. :)

Reply:
I agree with the original suggestion here. `TryXYZ` is a well-understood pattern that this clashes with. It's also used outside of Dictionaries even in the BCL, see `int.TryParse` and friends.


To avoid changing all of the existing callers of `FindMemoryType`, you can just reformulate it in terms of the Try variant (abridged):


```C#
uint FindMemoryType(...) 
{
    if (!TryFindMemoryType(..., out memType)) throw VeldridException();
    else return memType;
}
```